flycheck.el

```diff
@@ -1149,6 +1147,28 @@ Return the path of the file."
         tempfile)
     (flycheck-temp-file-system filename)))
 
+(defun flycheck-temp-directory (checker)
+  "Return the directory where CHECKER writes temporary files.
+
+Return nil if the CHECKER does not write temporary files."
+  (let ((args (flycheck-checker-arguments checker)))
+    (cond
+     ((memq 'source args) temporary-file-directory)
+     ((memq 'source-inplace args)
+      (if buffer-file-name (file-name-directory buffer-file-name)
+        temporary-file-directory))
+     (t nil))))
+
+(defun flycheck-temp-files-writable-p (checker)
+  "Whether CHECKER can write temporary files.
+
+If CHECKER has `source' or `source-inplace' in its `:command',
+return whether flycheck has the permissions to create the
+respective temporary files.
+
+Return t if CHECKER does not use temporary files."
+  (file-writable-p (flycheck-temp-directory checker)))
```

Comment:
Won't this raise an error if `(flycheck-temp-directory checker)` returns nil?

Reply:
Indeed!

So I could add a guard to `flycheck-temp-files-writable-p`, or I could just dispense with this function altogether and just use `flycheck-temp-directory` directly in the couple of places the former is called.

WDYT?

Reply:
I think a guard is fine :)

Reply:
Done.